service/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreateAppInsightsStep.java

```diff
@@ -0,0 +1,87 @@
+package bio.terra.landingzone.stairway.flight.create.resource.step;
+
+import bio.terra.landingzone.library.landingzones.definition.ArmManagers;
+import bio.terra.landingzone.library.landingzones.definition.ResourceNameGenerator;
+import bio.terra.landingzone.library.landingzones.definition.factories.ParametersResolver;
+import bio.terra.landingzone.library.landingzones.deployment.LandingZoneTagKeys;
+import bio.terra.landingzone.library.landingzones.deployment.ResourcePurpose;
+import bio.terra.landingzone.stairway.flight.LandingZoneFlightMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import com.azure.core.management.exception.ManagementException;
+import com.azure.resourcemanager.applicationinsights.models.ApplicationType;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CreateAppInsightsStep extends BaseResourceCreateStep {
+  private static final Logger logger = LoggerFactory.getLogger(CreateAppInsightsStep.class);
+  public static final String APP_INSIGHT_ID = "APP_INSIGHT_ID";
+
+  public CreateAppInsightsStep(
+      ArmManagers armManagers,
+      ParametersResolver parametersResolver,
+      ResourceNameGenerator resourceNameGenerator) {
+    super(armManagers, parametersResolver, resourceNameGenerator);
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext context) {
+    var appInsightId = context.getWorkingMap().get(APP_INSIGHT_ID, String.class);
+    try {
+      if (appInsightId != null) {
```

Comment:
This seems like a common pattern and a fair amount of boilerplate in each step, might be nice to extract to a base class method. The trick will be delegating the deletion call to the appropriate method, perhaps an abstract method  "deleteResource" that a base class method would invoke.

Reply:
Yes. That was my initial intent.